File: AtCoder/ABC/ABC157/B.c
#include<stdio.h>
int main(void){
int A[10]={0}, N=0, ch[10]={0};
for(int i=0;i<9;i++){
scanf("%d", &A[i]);
}
scanf("%d",&N);
int B[N];
for(int i=0;i<N;i++){
    scanf("%d", &B[i]);
}

for(int i=0;i<9;i++){
    for(int j=0;j<N;j++){
        if(A[i]==B[j]){
            ch[i]=1;
        }
    }
}
if(ch[0] && ch[1] && ch[2] == 1)printf("Yes\n");
else if(ch[3] && ch[4] && ch[5]== 1)printf("Yes\n");
else if(ch[6] && ch[7] && ch[8]== 1)printf("Yes\n");
else if(ch[0] && ch[3] && ch[6]== 1)printf("Yes\n");
else if(ch[1] && ch[4] && ch[7]== 1)printf("Yes\n");
else if(ch[2] && ch[5] && ch[8]== 1)printf("Yes\n");
else if(ch[0] && ch[4] && ch[8]== 1)printf("Yes\n");
else if(ch[2] && ch[4] && ch[6]== 1)printf("Yes\n");
else printf("No\n");
return 0;
}

File: AtCoder/ABC/ABC157/B.py
a = []
for i in range(3):
    a += list(map(int, input().split()))
n = int(input())
for _ in range(n):
    b = int(input())
    if b in a:
        i = a.index(b)
        a[i] = 0
if sum(a[0:3]) == 0 or sum(a[3:6]) == 0 or sum(a[6:]) == 0 \
    or sum(a[0::3]) == 0 or sum(a[1::3]) == 0 or sum(a[2::3]) == 0 \
    or sum(a[0::4]) == 0 or sum(a[2:7:2]) == 0:
    print("Yes")
else:
    print("No")